Document MSTEST0050 analyzer rule (#47608)
dotnet/docs: docs/core/testing/mstest-analyzers/mstest0050.md

ms.date: 07/29/2025
f1_keywords:
- MSTEST0050
- GlobalTestFixtureShouldBeValidAnalyzer
helpviewer_keywords:
- GlobalTestFixtureShouldBeValidAnalyzer
- MSTEST0050
author: Evangelink
ms.author: amauryleve
---
# MSTEST0050: Global test fixture should be valid

| Property                            | Value                                                                                    |
|-------------------------------------|------------------------------------------------------------------------------------------|
| **Rule ID**                         | MSTEST0050                                                                               |
| **Title**                           | Global test fixture should be valid                                                     |
| **Category**                        | Usage                                                                                    |
| **Fix is breaking or non-breaking** | Non-breaking                                                                             |
| **Enabled by default**              | Yes                                                                                      |

> | **Default severity**                | Warning                                                                                  |

Reviewer: ```suggestion
| **Default severity**                | Error                                                                                  |
```

| **Introduced in version**           | 3.10.0                                                                                   |

> | **Is there a code fix**             | Yes                                                                                      |

Reviewer: ```suggestion
| **Is there a code fix**             | No                                                                                       |
```

## Cause

A global test fixture method (marked with <xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyInitializeAttribute> or <xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyCleanupAttribute>) doesn't follow the required layout or has invalid configuration.

## Rule description

> Global test fixture methods must follow specific requirements to ensure proper test execution. This rule validates that methods marked with <xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyInitializeAttribute> or <xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyCleanupAttribute> adhere to the correct method signature and configuration rules.

Reviewer: ```suggestion
Global test fixture methods must follow specific requirements to ensure proper test execution. This rule validates that methods marked with `GlobalTestInitializeAttribute` or `GlobalTestCleanupAttribute` adhere to the correct method signature and configuration rules.
```

> Common violations include:
> 
> - Incorrect method signatures or parameters
> - Invalid access modifiers
> - Missing or incorrect return types
> - Improper method declarations

Reviewer: ```suggestion
The method must be `public`, `static`, non-generic, have a single parameter of type `TestContext`, and return `void` or `Task`. In addition, the containing type must be `public`, `static`, non-generic, and be marked with `TestClassAttribute`.
```
